test/com/google/enterprise/adaptor/database/TupleReaderTest.java

```diff
@@ -287,4 +421,103 @@ public void testBLOB() throws Exception {
+        + "<table>"
+        + "<table_rec>"
+        + "<COLNAME SQLType=\"BLOB\" ISNULL=\"true\"/>"
+        + "</table_rec>"
+        + "</table>"
+        + "</database>";
+    ResultSet rs = executeQueryAndNext("select * from data");
+    String result = generateXml(rs);
+    assertEquals(golden, result);
+  }
+
+  /**
+   * Test CLOB column.
+   */
+  @Test
+  public void testClob() throws Exception {
+    String clobData =
+        " Google's indices consist of information that has been"
+            + " identified, indexed and compiled through an automated"
+            + " process with no advance review by human beings. Given"
+            + " the enormous volume of web site information added,"
+            + " deleted, and changed on a frequent basis, Google cannot"
+            + " and does not screen anything made available through its"
+            + " indices. For each web site reflected in Google's"
+            + " indices, if either (i) a site owner restricts access to"
+            + " his or her web site or (ii) a site is taken down from"
+            + " the web, then, upon receipt of a request by the site"
+            + " owner or a third party in the second instance, Google"
+            + " would consider on a case-by-case basis requests to"
+            + " remove the link to that site from its indices. However,"
+            + " if the operator of the site does not take steps to"
+            + " prevent it, the automatic facilities used to create"
+            + " the indices are likely to find that site and index it"
+            + " again in a relatively short amount of time.";
+    executeUpdate("create table data(colname clob)");
+    String sql = "insert into data(colname) values (?)";
+    try (PreparedStatement ps = getConnection().prepareStatement(sql)) {
+      ps.setString(1, clobData);
+      assertEquals(1, ps.executeUpdate());
+    }
+    final String golden = ""
+        + "<database>"
+        + "<table>"
+        + "<table_rec>"
+        + "<COLNAME SQLType=\"CLOB\">"
+        + clobData
+        + "</COLNAME>"
+        + "</table_rec>"
+        + "</table>"
+        + "</database>";
+    ResultSet rs = executeQueryAndNext("select * from data");
+    String result = generateXml(rs);
+    assertEquals(golden, result);
+  }
+
+  @Test
+  public void testClob_null() throws Exception {
+    executeUpdate("create table data(colname clob)");
+    executeUpdate("insert into data(colname) values(null)");
+    final String golden = ""
+        + "<database>"
+        + "<table>"
+        + "<table_rec>"
+        + "<COLNAME SQLType=\"CLOB\" ISNULL=\"true\"/>"
+        + "</table_rec>"
+        + "</table>"
+        + "</database>";
+    ResultSet rs = executeQueryAndNext("select * from data");
+    String result = generateXml(rs);
+    assertEquals(golden, result);
+  }
+
+  @Test
+  public void testMultipleTypes() throws Exception {
+    executeUpdate("create table data(id integer, name varchar, "
```

Comment:
Break before these two string to avoid string concatenation (prefer higher-level breaks).